### PR TITLE
Add automatic 'delinquent reservations' finder and marker

### PR DESCRIPTION
**Pull request by ThePletch**

### CODE REVIEW

**Comment by Maelstroms** [comment]
It looks good to me

File: app/workers/some_class.rb [hunk 1]
```diff
@@ -1,5 +0,0 @@
-class SomeClass
```
Comment (ThePletch) [inline app/workers/some_class.rb]: i have no idea how this got here

File: app/views/late_reservation_mailer/delinquent_reservation.html.slim [hunk 2]
```diff
@@ -0,0 +1,7 @@
+h2 Your gifts were due today (#{Date.today.strftime("%B %d, %Y")}).
+p
+  | At least one of the gifts you reserved has not yet been received by Wediko.
+    If you shipped them recently, there's no reason to worry, but if you have yet to ship your gifts, please log in below to mark them as shipped.
```
Comment (caramcc) [inline app/views/late_reservation_mailer/delinquent_reservation.html.slim]: Should this also include a note about unreserving them? That way Wediko would have a bit more time to step in and order gifts the abandoned reservations

Comment (ThePletch) [inline app/views/late_reservation_mailer/delinquent_reservation.html.slim]: Now that you mention it, you can't actually mess with reservations once they're marked as delinquent. Let me fix that (and add a line to the email).

File: app/workers/delinquent_finder.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class DelinquentFinder
+  include Sidekiq::Worker
+  include Sidetiq::Schedulable
+
+  recurrence { daily }
+
+  def perform
+    late_unmarked_reservations = Reservation.joins(gift_request: {list: :event}).includes(donor: :user)
+                                   .where("events.end_date <= ?", Date.today)
+                                   .where(delinquent: false)
```
Comment (caramcc) [inline app/workers/delinquent_finder.rb]: Should this also have a clause that only checks for `:reserved` or `:shipped` reservations (i.e. "received" reservations aren't marked delinquent?) or is that handled somewhere else?

Comment (ThePletch) [inline app/workers/delinquent_finder.rb]: good call, will add

**Pull request merged**